### DISCUSSION TITLE
Use Coil3 in Compose with SVG

### DISCUSSION THREAD

**Discussion by StefanOltmann** [coil-kt/coil#2527] [Q&A] [answered]
I don't understand how Coil3 is supposed to be used for SVGs.

My goal is to display a SVG on Desktop & WASM.

I added these dependencies to `commonMain`:
```
implementation("io.coil-kt.coil3:coil-compose:3.0.0-alpha10")
implementation("io.coil-kt.coil3:coil-svg:3.0.0-alpha10")
```

I set the SVG factory in my `App()` composable:
```
setSingletonImageLoaderFactory {
    ImageLoader.Builder(PlatformContext.INSTANCE)
        .components {
            add(SvgDecoder.Factory())
        }
        .build()
}
```

But this displays always a black screen:
```
AsyncImage(
    model = mySvgString, // "<svg... and so on>"
    contentDescription = null
)
```

I also found for Coil2 this:
```
val model = ImageRequest.Builder(PlatformContext.INSTANCE)
.data(ByteBuffer.wrap(asteroid.biomesSVG.toByteArray()))
.decoderFactory(SvgDecoder.Factory())
.build()

AsyncImage(
    model = model,
    contentDescription = null
)
```
But there is no `ByteBuffer`.

What's the correct way to display SVGs (from a String) using Coil?

**Comment by colinrtwhite** [comment, marked as answer]
You should be able to pass a raw SVG as a `ByteArray`: `model = mySvgString.toByteArray()`.
